### PR TITLE
KAFKA-9738: Deprecate old Processor API

### PR DESCRIPTION
**Pull request by mjsax**
Implements KIP-1070, which deprecates:
 - Transformer
 - TransformerSupplier
 - ValueTransformer
 - ValueTransformerSupplier
 - MockProcessorContext

### CODE REVIEW

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -133,6 +133,16 @@ <h3 class="anchor-heading"><a id="streams_notable_changes" class="anchor-link"><
         More details about the new config <code>StreamsConfig#TOPOLOGY_OPTIMIZATION</code> can be found in <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-295%3A+Add+Streams+Configuration+Allowing+for+Optional+Topology+Optimization">KIP-295</a>.
     </p>
 
+    <h3><a id="streams_api_changes_400" href="#streams_api_changes_400">Streams API changes in 4.0.0</a></h3>
+
+    <p>
+        In previous release, a new version of the Processor API was introduces and the old Processor API was
```
Comment (bbejeck) [inline docs/streams/upgrade-guide.html]:
```suggestion
        In previous release, a new version of the Processor API was introduced and the old Processor API was
```

File: streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java [hunk 2]
```diff
@@ -1,1150 +0,0 @@
...
-                "--property", "key.deserializer.window.size.ms=500",
-            };
-
-            ConsoleConsumer.run(new ConsoleConsumerOptions(args));
-            newStream.flush();
-            System.setOut(originalStream);
-            return newConsole.toString();
-        }
-    }
-}
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationIntegrationTest.java]: ~Why delete this test? I don't have an issue, just wanting to understand.  I have the same comment for the other deleted integration tests.~

NM after reading the commit message again and reviewing other related PRs I get what's going on.

**Review by bbejeck — COMMENTED** [review]
Thanks for the PR @mjsax this LGTM, I just have a couple of minor comments.
I'll make another pass and give an official approval when it's ready for merging.

**Comment by mjsax** [comment]
@bbejeck Rebased this PR. It should be clean now. Hope I did not miss anything during rebasing...

**Review by bbejeck — APPROVED** [review]
LGTM

**Pull request merged**